snippets/rust.json

```diff
@@ -36,6 +36,20 @@
         ],
         "description": "Insert println!"
     },
+    "assert": {
+		"prefix": "assert",
+		"body": [
+			"assert!($1)$0"
```

Comment:
The syntax here and for assert_eq looks wrong, should it be `assert!($1, $0)`? Also I don't see why it is in this order and not `$0, $1`.

Reply:
Hi @nrc, [the `$0` tabstop in a snippet actually denotes the final cursor position](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax). Its purpose is to allow user to 'jump out' of the braces on the final 'tab' keypress. It's very convenient because it allows users to add a semicolon at the end without needing to manually move the cursor there.

But it's not only added there for convenience reasons, it's also to have parity with function snippet completions. If you're familiar with them, they all have a final tabstop immediately to the right of the closing brace.

Reply:
Thanks for the explanation!